reflect changes w.r.t the new api

Freemanlabs opened the pull request.

Freemanlabs committed:
reflect changes w.r.t the new api

dnk8n commented:
Thanks @Freemanlabs 

Does the 'Details' link under the checks section work for you?

If not, here is a copy of the build logs (TypeError: Cannot read property 'length' of undefined):

```
23:15:52.563  	Cloning github.com/dnk8n/demo-cn (Branch: api, Commit: ddb9f70)
23:15:53.239  	Cloning completed: 675.634ms
23:15:53.289  	Analyzing source code...
23:15:54.710  	Installing build runtime...
23:15:56.813  	Build runtime installed: 2.103s
23:15:59.370  	Looking up build cache...
23:16:00.639  	Build cache downloaded [12.12 MB]: 931.346ms
23:16:01.986  	Installing dependencies...
23:16:04.260  	npm WARN bootstrap@5.0.0-beta3 requires a peer of @popperjs/core@^2.9.1 but none is installed. You must install peer dependencies yourself.
23:16:04.272  	npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
23:16:04.273  	npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
23:16:04.274  	up to date in 1.725s
23:16:04.395  	45 packages are looking for funding
23:16:04.396  	  run `npm fund` for details
23:16:04.414  	Detected Next.js version: 10.2.0
23:16:04.416  	Running "npm run build"
23:16:04.684  	> demo-cn@0.1.0 build /vercel/path0/portal
23:16:04.684  	> next build
23:16:05.198  	info  - Using webpack 5. Reason: no next.config.js https://nextjs.org/docs/messages/webpack5
23:16:05.340  	info  - Checking validity of types...
23:16:05.396  	info  - Creating an optimized production build...
23:16:12.357  	<i> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Node.js doesn't offer a (nice) way to introspect the ESM dependency graph yet.
23:16:12.358  	<i> Until a full solution is available webpack uses an experimental ESM tracking based on parsing.
23:16:12.358  	<i> As best effort webpack parses the ESM files to guess dependencies. But this can lead to expensive and incorrect tracking.
23:16:17.506  	<i> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Node.js doesn't offer a (nice) way to introspect the ESM dependency graph yet.
23:16:17.506  	<i> Until a full solution is available webpack uses an experimental ESM tracking based on parsing.
23:16:17.507  	<i> As best effort webpack parses the ESM files to guess dependencies. But this can lead to expensive and incorrect tracking.
23:16:19.397  	<w> [webpack.cache.PackFileCacheStrategy] Serializing big strings (181kiB) impacts deserialization performance (consider using Buffer instead and decode when needed)
23:16:19.397  	<w> [webpack.cache.PackFileCacheStrategy] Serializing big strings (152kiB) impacts deserialization performance (consider using Buffer instead and decode when needed)
23:16:19.507  	info  - Compiled successfully
23:16:19.507  	info  - Collecting page data...
23:16:19.994  	> Build error occurred
23:16:19.996  	TypeError: Cannot read property 'length' of undefined
23:16:19.996  	    at buildFundingSchema (/vercel/path0/portal/.next/server/pages/[page].js:3212:19)
23:16:19.997  	    at getStaticPaths (/vercel/path0/portal/.next/server/pages/[page].js:2690:93)
23:16:19.997  	    at buildStaticPaths (/vercel/path0/portal/node_modules/next/dist/build/utils.js:16:86)
23:16:19.997  	    at /vercel/path0/portal/node_modules/next/dist/build/utils.js:26:628
23:16:19.998  	    at processTicksAndRejections (internal/process/task_queues.js:95:5)
23:16:19.998  	    at async Span.traceAsyncFn (/vercel/path0/portal/node_modules/next/dist/telemetry/trace/trace.js:6:584) {
23:16:19.998  	  type: 'TypeError'
23:16:19.998  	}
23:16:20.025  	npm ERR! code ELIFECYCLE
23:16:20.026  	npm ERR! errno 1
23:16:20.030  	npm ERR! demo-cn@0.1.0 build: `next build`
23:16:20.030  	npm ERR! Exit status 1
23:16:20.030  	npm ERR! 
23:16:20.030  	npm ERR! Failed at the demo-cn@0.1.0 build script.
23:16:20.031  	npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
23:16:20.039  	npm ERR! A complete log of this run can be found in:
23:16:20.039  	npm ERR!     /vercel/.npm/_logs/2021-10-05T20_16_20_030Z-debug.log
23:16:20.055  	Error: Command "npm run build" exited with 1
```

dnk8n committed:
Commit changes

dnk8n committed:
Merge pull request #2 from dnk8n/api-dean-updates

Api dean updates

Freemanlabs committed:
fix: change header bg image and title text

dnk8n committed:
Fix language filter, given that it now is a single language per item

dnk8n committed:
Put back what Charles had already done. Works now that only single language per item. I also ensure that null sort values come last

Freemanlabs committed:
revert: switch language implemetation to ddb9f702

Freemanlabs committed:
Merge branch 'api' of https://github.com/dnk8n/demo-cn into api
merge: pull remote changes to local

dnk8n committed:
Add click through pages

Freemanlabs committed:
Merge branch 'api' of https://github.com/dnk8n/demo-cn into api
merge: pull remote changes to local

Freemanlabs committed:
fix: make detail page display right content

detail page should display the right content according to the link specified in the url

dnk8n committed:
Take out hyperlink to research db docs
